spec/models/pawn_spec.rb

```diff
@@ -62,8 +62,8 @@
 
     it 'should return false if moving one space forward when destination square is occupied' do
       pawn = FactoryGirl.create(:pawn, x_position: 4, y_position: 4, is_white: false)
-      FactoryGirl.create(:rook, x_position: 4, y_position: 5, is_white: true)
-      expect(pawn.valid_move?(4, 5)).to eq false
+      FactoryGirl.create(:rook, x_position: 4, y_position: 3, is_white: true)
+      expect(pawn.valid_move?(4, 3)).to eq false
     end
 
```

Comment:
This is still incorrect. White pawn start at bottom of board, so forward moves will result in y coordinate increasing. In this case, we are testing to see if piece can be moved from (4, 3), into (4, 4), where (4, 4) is occupied by a different piece the pawn cannot capture.

Reply:
If you look at travis test results, giving wrong coor is also reason this test is failing

Reply:
I'm pretty sure we're seeing if a black pawn can be moved from (4, 4) to (4, 3), where (4, 3) is currently occupied by a white rook. This would be a forward move for the black pawn. I believe the reason this test was failing was because of incomplete valid_move? logic, and because the pawn and rook weren't related to each other by being part of the same game instance (I discovered that second part when I went into pry).

I have since fixed it, and the tests now pass.

Reply:
(see the detailed commit message)